### ISSUE TITLE
supernova segfaults loading cmt.so via LadspaUGen_supernova.so

### ISSUE THREAD

**Issue by TauPan** [KXStudio/Repository#258]
I've got supercollider and sc3-plugins compiled from recent master/devel and the cmt package from the kxstudio repos on an ubuntu-studio system.

When I start supernova, I get a segmentation fault in LadspaUGen_supernova.so in strlen:
~~~gdb
Thread 1 "supernova" received signal SIGSEGV, Segmentation fault.                                                                                                             
strlen () at ../sysdeps/x86_64/multiarch/../strlen.S:120                                                                                                                      
120     ../sysdeps/x86_64/multiarch/../strlen.S: Datei oder Verzeichnis nicht gefunden.                                                                                       
(gdb) w                                                                                                                                                                       
Missing arguments.                                                                                                                                                            
(gdb) where                                                                                                                                                                   
#0  strlen () at ../sysdeps/x86_64/multiarch/../strlen.S:120                                                                                                                  
#1  0x00007ffff7fe7cd5 in __GI__dl_exception_create (exception=0x7ffffff26f70, objname=0x2f61707364616c2f <error: Cannot access memory at address 0x2f61707364616c2f>,        
    errstring=0x7ffff7ff4b69 "shared object not open") at dl-exception.c:74                                                                                                   
#2  0x00007ffff78a1808 in __GI__dl_signal_error (errcode=0, objname=<optimized out>, occation=<optimized out>, errstring=<optimized out>) at dl-error-skeleton.c:117          
#3  0x00007ffff7fe6f51 in _dl_close (_map=0x555555b37fa0) at dl-close.c:856                                                                                                   
#4  0x00007ffff78a18b8 in __GI__dl_catch_exception (exception=exception@entry=0x7ffffff26f70, operate=<optimized out>, args=<optimized out>) at dl-error-skeleton.c:208       
#5  0x00007ffff78a1983 in __GI__dl_catch_error (objname=0x55555593ff20, errstring=0x55555593ff28, mallocedp=0x55555593ff18, operate=<optimized out>, args=<optimized out>)    
    at dl-error-skeleton.c:227                                                                                                                                                
#6  0x00007ffff7f62b59 in _dlerror_run (operate=operate@entry=0x7ffff7f62420 <dlclose_doit>, args=0x555555b37fa0) at dlerror.c:170                                            
#7  0x00007ffff7f62468 in __dlclose (handle=<optimized out>) at dlclose.c:46                                                                                                  
#8  0x00007fffef8d0a48 in LADSPAPluginSearch () from /usr/local/stow/supercollider/lib/SuperCollider/plugins/LadspaUGen_supernova.so                                          
#9  0x00007fffef8d0740 in load () from /usr/local/stow/supercollider/lib/SuperCollider/plugins/LadspaUGen_supernova.so                                                        
#10 0x00005555555ec987 in nova::sc_ugen_factory::load_plugin(boost::filesystem::path const&) ()                                                                               
#11 0x00005555555ecd40 in nova::sc_ugen_factory::load_plugin_folder(boost::filesystem::path const&) ()                                                                        
#12 0x0000555555585372 in supernova_main(int, char**) ()                                                                                                                      
#13 0x00007ffff77650b3 in __libc_start_main (main=0x555555580ab0 <main>, argc=1, argv=0x7fffffffcaa8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>,  
    stack_end=0x7fffffffca98) at ../csu/libc-start.c:308                                                                                                                      
#14 0x00005555555835ce in _start ()                
~~~

When I do an strace, the last file loaded with `openat` is cmt.so:
~~~strace
[pid 3373730] mprotect(0x7f44c0449000, 4096, PROT_READ) = 0               
[pid 3373730] openat(AT_FDCWD, "/usr/lib/ladspa/cmt.so", O_RDONLY|O_CLOEXEC) = 9
[pid 3373730] read(9, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\3400\0\0\0\0\0\0"..., 832) = 832
[pid 3373730] fstat(9, {st_mode=S_IFREG|0644, st_size=104672, ...}) = 0
[pid 3373730] mmap(NULL, 2199848, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 9, 0) = 0x7f44c002e000
[pid 3373730] mprotect(0x7f44c0046000, 2093056, PROT_NONE) = 0
[pid 3373730] mmap(0x7f44c0245000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 9, 0x17000) = 0x7f44c0245000
[pid 3373730] close(9)                  = 0 
[pid 3373730] mprotect(0x7f44c0245000, 8192, PROT_READ) = 0
[pid 3373730] --- SIGSEGV {si_signo=SIGSEGV, si_code=SI_KERNEL, si_addr=NULL} ---
[pid 3373742] <... poll resumed> <unfinished ...>) = ?
[pid 3373741] <... epoll_wait resumed> <unfinished ...>) = ?
[pid 3373740] <... futex resumed>)      = ? 
[pid 3373737] <... futex resumed>)      = ? 
[pid 3373740] +++ killed by SIGSEGV (core dumped) +++
[pid 3373737] +++ killed by SIGSEGV (core dumped) +++
[pid 3373741] +++ killed by SIGSEGV (core dumped) +++
[pid 3373742] +++ killed by SIGSEGV (core dumped) +++
+++ killed by SIGSEGV (core dumped) +++  
zsh: segmentation fault (core dumped)  strace -fff supernova
~~~

When I replace the `cmt` package from kxstudio (5:1.17-1kxstudio1) with the one from ubuntu-studio (1.17-1build1) the segfault disappears.

Not sure if you support this kind of thing (self-compiled supercollider, kxstudio mixed with ubuntu-studio) so simply close if not. (And the fix is easy enough.) This issue can still serve as a hint to the astute googler after it's been closed. :-)

**Comment by falkTX** [comment]
Thanks for the report.
I accept all kinds of issues, as long as they are my fault or something I can fix.
The CMT thing seems like my fault indeed. Let me try to get it fixed ASAP

**Comment by falkTX** [comment]
Done, could easily reproduce the issue.
Fix coming in 1.17-1kxstudio2 package update.
